NoQuD/StepCharacteristicSolve/StepCharacteristicSolver.py

```diff
@@ -79,21 +115,21 @@ def __init__(self, sig_t, sig_s_in, sig_s_out, sig_f, nu, chi, ab = ab_default,
                 for i in xrange(10):
                     if i + 1 <= len(self.ab) / 2 and j == self.core_mesh_length:
                         self.angular_flux_edge[k][j][i] = self.phi_R_old[k][i]
-                    elif i + 1 > len(ab) / 2 and j == 0:
+                    elif i + 1 > len(self.ab) / 2 and j == 0:
                         self.angular_flux_edge[k][j][i] = self.phi_L_old[k][i - 5]
 
     # With a given flux, the source from scattering is calculated.
     def form_scatter_source(self):
         # Form scattering source.
-        self.spatial_sig_s_out = numpy.zeros([self.groups, self.core_mesh_length])
+        self.spatial_sig_s_out = numpy.zeros((self.groups, self.core_mesh_length), dtype=numpy.float64)
         for i in xrange(self.core_mesh_length):
             for k in xrange(self.groups):
                 self.spatial_sig_s_out[1 - k][i] = self.dx * self.flux_old[k][i] * self.sig_s_out[k][self.material[i]]
 
     # With a given flux, the source from fission is calculated.
     def form_fission_source(self):
         # Form fission source.
-        self.spatial_fission_new = numpy.zeros([self.groups, self.core_mesh_length])
+        self.spatial_fission_new = numpy.zeros((self.groups, self.core_mesh_length), dtype=numpy.float64)
         for i in xrange(self.core_mesh_length):
```

Comment:
This may be a lot of work, but it might be worth it to change you iteration variable (i,j,k, etc.) to a variable the iteration is actually going over. For example, if 'i' is iterating over the x-direction, you could change 'i' to 'x'. It would create an easier to read code. However, that may be too much work to do at this moment.

Reply:
That would greatly improve readability. Once all the major functionality is built in, I may return and do this.